src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2572,7 +2572,14 @@ impl Coordinator {
         let id_bundle = self
             .index_oracle(cluster_id)
             .sufficient_collections(&depends_on);
-        let timeline = self.validate_timeline_context(id_bundle.iter())?;
+        let mut timeline = self.validate_timeline_context(id_bundle.iter())?;
+        if matches!(timeline, TimelineContext::TimestampIndependent)
+            && (from.contains_temporal() || when.advance_to_timestamp().is_some())
+        {
+            // If the from IDs are timestamp independent but the query contains temporal functions,
+            // or uses AS OF then the timeline context needs to be upgraded to timestamp dependent.
+            timeline = TimelineContext::TimestampDependent;
+        }
```

Comment:
Should this include `AS OF` queries without temporal filters? I'm not sure.

What should the timestamp be of the following query?

```
SUBSCRIBE (SELECT 1) AS OF 42
----
<timestamp>  1  1
```
??

Reply:
@frankmcsherry curious what you're thoughts are here.